rusk/tests/services/multi_transfer.rs

```diff
@@ -0,0 +1,613 @@
+    TestStore.retrieve_ssk(2).expect("Should not fail in test")
+});
+
+fn generate_notes(rusk: &mut Rusk) -> Result<()> {
+    info!("Generating a note");
+    let mut rng = StdRng::seed_from_u64(0xdead);
+
+    let psk_0 = SSK_0.public_spend_key();
+    let psk_1 = SSK_1.public_spend_key();
+    let psk_2 = SSK_2.public_spend_key();
+
+    let note_0 = Note::transparent(&mut rng, &psk_0, INITIAL_BALANCE);
+    let note_1 = Note::transparent(&mut rng, &psk_1, INITIAL_BALANCE);
+    let note_2 = Note::transparent(&mut rng, &psk_2, INITIAL_BALANCE);
+
+    let mut rusk_state = rusk.state()?;
+    let mut transfer = rusk_state.transfer_contract()?;
+
+    transfer.push_note(BLOCK_HEIGHT, note_0)?;
+    transfer.push_note(BLOCK_HEIGHT, note_1)?;
+    transfer.push_note(BLOCK_HEIGHT, note_2)?;
+
+    transfer.update_root()?;
+
+    info!("Updating the new transfer contract state");
+    unsafe {
+        rusk_state
+            .set_contract_state(&rusk_abi::transfer_contract(), &transfer)?;
+    }
+
+    rusk_state.finalize();
+
+    Ok(())
+}
+
+/// Executes three different transactions in the same block, expecting only two
+/// to be included due to exceeding th block gas limit
+fn wallet_transfer(
+    wallet: &wallet::Wallet<TestStore, TestStateClient, TestProverClient>,
+    channel: tonic::transport::Channel,
+    amount: u64,
+) {
+    // Sender psk
+    let psk_0 = SSK_0.public_spend_key();
+    let psk_1 = SSK_1.public_spend_key();
+    let psk_2 = SSK_2.public_spend_key();
+
+    let refunds = vec![psk_0, psk_1, psk_2];
+
+    // Generate a receiver psk
+    let receiver = wallet
+        .public_spend_key(3)
+        .expect("Failed to get public spend key");
+
+    let mut rng = StdRng::seed_from_u64(0xdead);
+    let nonce = BlsScalar::random(&mut rng);
+
+    let initial_balance_0 =
+        wallet.get_balance(0).expect("Failed to get the balance");
+    let initial_balance_1 =
+        wallet.get_balance(1).expect("Failed to get the balance");
+    let initial_balance_2 =
+        wallet.get_balance(2).expect("Failed to get the balance");
+
+    // Check the senders initial balance is correct
+    assert_eq!(
+        initial_balance_0, INITIAL_BALANCE,
+        "Wrong initial balance for the sender"
+    );
+    assert_eq!(
+        initial_balance_1, INITIAL_BALANCE,
+        "Wrong initial balance for the sender"
+    );
+    assert_eq!(
+        initial_balance_2, INITIAL_BALANCE,
+        "Wrong initial balance for the sender"
+    );
+
+    // Check the receiver initial balance is zero
+    assert_eq!(
+        wallet.get_balance(3).expect("Failed to get the balance"),
+        0,
+        "Wrong initial balance for the receiver"
+    );
+
+    let mut txs = Vec::with_capacity(3);
+
+    for i in 0..3 {
+        let tx = wallet
+            .transfer(
+                &mut rng,
+                i,
+                &refunds[i as usize],
+                &receiver,
+                amount,
+                1_000_000_000,
```

Comment:
Does this mean that is now safe to send txs even with GasLimit higher than the BlockGasLimit?

Reply:
Yes it does. As long as the gas spent is ultimately smaller or equal than the gas left in the block it will pass.